src/projection.jl

```diff
@@ -1,14 +1,10 @@
 function projection(d::Integer, q::Integer, ::Type{elType}=typeof(1.0)) where {elType}
-    # Proj(deriv) = kron(diagm(0 => ones(elType, d)), [i==(deriv+1) ? 1 : 0 for i in 1:q+1]')
-
-    # Slightly faster version of the above:
-    D = d * (q + 1)
     Proj(deriv) = begin
-        P = zeros(elType, d, D)
-        @simd ivdep for i in deriv*d+1:D+1:d*D
-            @inbounds P[i] = 1
+        e_i = zeros(q+1, 1)
+        if deriv <= q
```

Comment:
I think this is only there to handle the creation of `E2` in caches.jl for the case where we use an IWP1 prior as in this case there is no projection matrix, right? If yes, maybe make this more explicit somehow. At least throw a warning or something.